frontend/app-development/hooks/mutations/useAddLayoutSetMutation.ts

```diff
@@ -1,25 +1,38 @@
 import { useMutation, useQueryClient } from '@tanstack/react-query';
 import { useServicesContext } from 'app-shared/contexts/ServicesContext';
 import { QueryKey } from 'app-shared/types/QueryKey';
-import type { LayoutSetConfig } from 'app-shared/types/api/LayoutSetsResponse';
+import type { LayoutSetConfig, LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
 import { useLocalStorage } from 'app-shared/hooks/useLocalStorage';
 
 export const useAddLayoutSetMutation = (org: string, app: string) => {
   const { addLayoutSet } = useServicesContext();
   const queryClient = useQueryClient();
   const [_, setSelectedLayoutSet] = useLocalStorage<string>('layoutSet/' + app, null);
 
+  const isLayoutSets = (obj: any): obj is LayoutSets => {
```

Comment:
Is it possible that we use another type than ```any``` ? 😄

Reply:
In theory I guess, but since the endpoint can return both a 200 OK with the layoutsets and a 200 OK with a informal message of what why the layout set was not added, the object to test in this method can be both layoutSets and a keyValuePair 🤔 I dont know what best practice is in this case

Reply:
Maybe we can define a type where the body can be either a layoutset or a message? 
And then perform some checks to see which type it is before proceeding. 
Maybe we can use something like this 🤔 
```tsx
type InputObject = LayoutSets | string;

const isLayoutSets = (obj: InputObject): obj is LayoutSets => {
    return typeof obj === 'object' && obj !== null && 'sets' in obj;
};
```

Reply:
Or actually, I tried now, but facing some typescript trouble with the type coming in from the query mutation response, which is void 🤔

Reply:
This was fixed by adding a return type to the ```addLayoutSet``` function 😄